src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs

```diff
@@ -938,9 +939,15 @@ internal static bool RequiresExplicitOverride(this MethodSymbol method, out bool
             if (!methodimplWouldBeAmbiguous)
                 return true;
 
+            Debug.Assert(runtimeOverriddenMethod is { });
+
             // We produce the warning when a methodimpl would be required but would be ambiguous to the runtime.
-            // This ambiguity only arises due to collision after generic substitution.
-            warnAmbiguous = csharpOverriddenMethod.ContainingType.IsGenericType;
+            // However, if there was a duplicate definition for the runtime signature of the overridden
+            // method where it was originally delcared, that would have been an error.  In that case we suppress
+            // the warning as a cascaded diagnostic.
+            bool originalOverriddenMethodWasAmbiguious =
+                csharpOverriddenMethod.IsDefinition || csharpOverriddenMethod.OriginalDefinition.MethodHasRuntimeCollisionInSubstututedForm();
```

Comment:
>MethodHasRuntimeCollisionInSubstututedForm [](start = 97, length = 42)

Use of this method is confusing because original definition couldn't have substituted form.